### PR TITLE
Add account validation module

### PR DESCRIPTION
**Pull request by a-leonardi**
Added an AccountValidator class that can be registered when instaciating
a module in the same way as triggers or actions
This class handles the account validation process, calling an abstract
validator method that the user will have to fill in himself, and sending
the return value of validator to a Symphony callback endpoint

Parent task: [#3](https://github.com/SekoiaLab/custom-integrations/issues/3)

### CODE REVIEW

File: sekoia_automation/module.py [hunk 1]
```diff
@@ -269,6 +269,11 @@ def register(self, item: type["ModuleItem"], name: str = ""):
             item.name = name
         self._items[name] = item
 
+    def register_account_validator(self):
+        from sekoia_automation.account_validator import AccountValidator
+
+        self.register(AccountValidator, "validate_module_configuration")
```
Comment (Darkheir) [inline sekoia_automation/module.py:275]: The creator of the module must be able to specify its own implementation of the validator

```suggestion
    def register_account_validator(self, validator: type["AccountValidator"] ):
        self.register(validator, "validate_module_configuration")
```

File: sekoia_automation/account_validator.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+        Returns:
+            bool: True if the module is valid, False otherwise
+        """
+
+    def execute(self):
+        """Validates the account (module_configuration) of the module
+        and sends the result to Symphony."""
+        # Call the actual validation procedure
+        status = self.validator()
```
Comment (Darkheir) [inline sekoia_automation/account_validator.py:23]: Before running the validation we must retrieve the secrets from the API the same way the actions do it.

File: sekoia_automation/account_validator.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            self.VALIDATION_CALLBACK_URL_FILE_NAME
+        )
+        response = requests.request(
+            "POST",
+            validation_callback_url,
+            json=data,
+            headers={"Authorization": f"Bearer {self.token}"},
+            timeout=30,
+        )
+        response.raise_for_status()
```
Comment (Darkheir) [inline sekoia_automation/account_validator.py:37]: You should be able to rely on the default `_send_request` method. In order for it to work we juste have to use the same callback file for validators and actions.
Your PR on the action worker does it, so it should work :)

```suggestion
        self._send_request(data)
```

File: sekoia_automation/account_validator.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import requests
+
+from sekoia_automation.module import ModuleItem
+
+
+class AccountValidator(ModuleItem):
+    VALIDATION_CALLBACK_URL_FILE_NAME = "validation_callback_url"
+
+    @abstractmethod
+    def validator(self) -> bool:
```
Comment (Darkheir) [inline sekoia_automation/account_validator.py:12]:
```suggestion
    def validate(self) -> bool:
```

File: sekoia_automation/account_validator.py (hunk 2 above)
Comment (Darkheir) [inline sekoia_automation/account_validator.py:23]:
```suggestion
        status = self.validate()
```

**Review by Darkheir — CHANGES_REQUESTED** [review]

File: sekoia_automation/account_validator.py (hunk 2 above)
Comment (a-leonardi) [reply sekoia_automation/account_validator.py]: I don't get it, why do we need to do that?

Comment (Darkheir) [reply sekoia_automation/account_validator.py]: The secrets are not passed to the action worker so we need a way to retrieve them.

**Review by Darkheir — APPROVED** [review]

**Review by Darkheir — APPROVED** [review]

**Pull request merged**